ait/commons/util/command/upload.py

```diff
@@ -38,7 +89,7 @@ def run(self):
                     ps.append(p)
 
             # create list of files to upload
-            fs = []
+            files = []
```

Comment:
Maybe use a `set`. I assume this should be a unique list?

Reply:
if all file names are unique, then yes. I'm not sure if the tool can rely on always rely on that being the case though

Reply:
Can filenames be non-unique in a filesystem?

Reply:
with nested directories, yes

Reply:
But then they still have to be unique in AWS right? I don't see any renaming logic

Reply:
That's true. And this just made me realise that have the existing file check as part of `upload_files` only checks for files that may be there in the bucket, before an upload run starts. The check should in the the `upload_file` function instead.

Coming back to your comment, if the list does have duplicate files, then the action should be to inform the user about the duplicity rather than silently removing it, so that they know. If two files have the same names, and one gets uploaded before the other in the same run, the user should know why the second file was not uploaded. This would also save us potential debugging effort, if this case ever arises, i.e. a count mismatch between the total number of files in the user directory vs the ones uploaded to s3.

Reply:
Agreed and agreed.

Sets would still be useful here. Checking if an item exists in a set is a lot quicker than checking if an item exists in a list. `item in my_set` is O(1) whereas `item in my_list` is O(n)

Reply:
agreed on sets being faster, but as long as we are supporting getting files from nested subdirectories, and we could potentially have duplicate file names, I want us to inform the user about the duplicity rather than silently removing them using sets. So let's stick to lists for now